[Bug] fail to query msgs when tls is enabled.

### Before Creating the Bug Report

- [X] I found a bug, not just asking a question, which should be created in [GitHub Discussions](https://github.com/apache/rocketmq/discussions).

- [X] I have searched the [GitHub Issues](https://github.com/apache/rocketmq/issues) and [GitHub Discussions](https://github.com/apache/rocketmq/discussions)  of this repository and believe that this is not a duplicate.

- [X] I have confirmed that this bug belongs to the current repository, not other repositories of RocketMQ.


### Runtime platform environment

ubuntu

### RocketMQ version

5.2.1-snapshot

### JDK Version

_No response_

### Describe the Bug

query messag by id fail after open tls



### Steps to Reproduce

query message:
```
JAVA_OPT=-Dtls.enable=true bin/mqadmin queryMsgByuniqueKey -n localhost:9876 -t  topic -i 644F006A17637B8C77617F487C310002
```
works fine without `JAVA_OPT=-Dtls.enable=true`

### What Did You Expect to See?

query msg success

### What Did You See Instead?

query msg fail
```
JAVA_OPT=-Dtls.enable=true bin/mqadmin queryMsgByuniqueKey -n localhost:9876 -t  topic1 -i AC11000144842EF1E4FA87ECFE6E0000
org.apache.rocketmq.tools.command.SubCommandException: QueryMsgByUniqueKeySubCommand command failed
        at org.apache.rocketmq.tools.command.message.QueryMsgByUniqueKeySubCommand.execute(QueryMsgByUniqueKeySubCommand.java:202)
        at org.apache.rocketmq.tools.command.MQAdminStartup.main0(MQAdminStartup.java:176)
        at org.apache.rocketmq.tools.command.MQAdminStartup.main(MQAdminStartup.java:126)
Caused by: org.apache.rocketmq.client.exception.MQClientException: CODE: 208  DESC: query message by key finished, but no message.
For more information, please visit the url, https://rocketmq.apache.org/docs/bestPractice/06FAQ
        at org.apache.rocketmq.client.impl.MQAdminImpl.queryMessage(MQAdminImpl.java:465)
        at org.apache.rocketmq.client.impl.MQAdminImpl.queryMessage(MQAdminImpl.java:317)
        at org.apache.rocketmq.client.impl.MQAdminImpl.queryMessageByUniqKey(MQAdminImpl.java:286)
        at org.apache.rocketmq.tools.admin.DefaultMQAdminExtImpl.queryMessageByUniqKey(DefaultMQAdminExtImpl.java:1777)
        at org.apache.rocketmq.tools.admin.DefaultMQAdminExt.queryMessageByUniqKey(DefaultMQAdminExt.java:791)
        at org.apache.rocketmq.tools.command.message.QueryMsgByUniqueKeySubCommand.queryById(QueryMsgByUniqueKeySubCommand.java:66)
        at org.apache.rocketmq.tools.command.message.QueryMsgByUniqueKeySubCommand.execute(QueryMsgByUniqueKeySubCommand.java:199)
        ... 2 more
```
work fine without tls:
```
bin/mqadmin queryMsgByuniqueKey -n localhost:9876 -t  topic1 -i AC11000144842EF1E4FA87ECFE6E0000
Topic:               topic1
Tags:                [null]
Keys:                [null]
Queue ID:            3
Queue Offset:        1092102
CommitLog Offset:    47211240955
Reconsume Times:     0
Born Timestamp:      2024-05-27 09:27:35,792
Store Timestamp:     2024-05-27 09:27:35,811
Born Host:           172.28.198.133:43460
Store Host:          172.28.198.133:10911
System Flag:         0
Properties:          {MSG_REGION=DefaultRegion, UNIQ_KEY=AC11000144842EF1E4FA87ECFE6E0000, CLUSTER=DefaultCluster, WAIT=true, TRACE_ON=true}
Message Body Path:   /tmp/rocketmq/msgbodys/AC11000144842EF1E4FA87ECFE6E0000


WARN: No Consumer
```

### Additional Context

happens after https://github.com/apache/rocketmq/issues/7757

--- a/remoting/src/main/java/org/apache/rocketmq/remoting/netty/FileRegionEncoder.java
+++ b/remoting/src/main/java/org/apache/rocketmq/remoting/netty/FileRegionEncoder.java
@@ -18,9 +18,6 @@
 package org.apache.rocketmq.remoting.netty;
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.ByteBufAllocator;
-import io.netty.buffer.CompositeByteBuf;
-import io.netty.buffer.Unpooled;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.channel.FileRegion;
 import io.netty.handler.codec.MessageToByteEncoder;
@@ -54,12 +51,9 @@ public class FileRegionEncoder extends MessageToByteEncoder<FileRegion> {
         WritableByteChannel writableByteChannel = new WritableByteChannel() {
             @Override
             public int write(ByteBuffer src) {
-                // To prevent mem_copy.
-                CompositeByteBuf b = (CompositeByteBuf) out;
-                // Have to increase writerIndex manually.
-                ByteBuf unpooled = Unpooled.wrappedBuffer(src);
-                b.addComponent(true, unpooled);
-                return unpooled.readableBytes();
+                int prev = out.writerIndex();
+                out.writeBytes(src);
+                return out.writerIndex() - prev;
             }
 
             @Override
@@ -81,11 +75,5 @@ public class FileRegionEncoder extends MessageToByteEncoder<FileRegion> {
             }
             msg.transferTo(writableByteChannel, transferred);
         }
-    }
-
-    @Override
-    protected ByteBuf allocateBuffer(ChannelHandlerContext ctx, FileRegion msg, boolean preferDirect) throws Exception {
-        ByteBufAllocator allocator = ctx.alloc();
-        return preferDirect ? allocator.compositeDirectBuffer() : allocator.compositeHeapBuffer();
     }
 }

--- a/remoting/src/test/java/org/apache/rocketmq/remoting/netty/FileRegionEncoderTest.java
+++ b/remoting/src/test/java/org/apache/rocketmq/remoting/netty/FileRegionEncoderTest.java
@@ -21,15 +21,14 @@ import io.netty.buffer.ByteBuf;
 import io.netty.channel.DefaultFileRegion;
 import io.netty.channel.FileRegion;
 import io.netty.channel.embedded.EmbeddedChannel;
-import org.junit.Assert;
-import org.junit.Test;
-
 import java.io.BufferedOutputStream;
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.util.Random;
 import java.util.UUID;
+import org.junit.Assert;
+import org.junit.Test;
 
 public class FileRegionEncoderTest {
 

